src/globus_cli/tokenstore.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+
+import globus_sdk
+from globus_sdk.tokenstorage import SQLiteAdapter
+
+from .config import invalidate_old_config
+
+# internal constants
+_CLIENT_DATA_CONFIG_KEY = "auth_client_data"
+
+# env vars used throughout this module
+GLOBUS_ENV = os.environ.get("GLOBUS_SDK_ENVIRONMENT")
+GLOBUS_PROFILE = os.environ.get("GLOBUS_PROFILE")
+
+
+def _template_client_id():
+    template_id = "95fdeba8-fac2-42bd-a357-e068d82ff78e"
+    if GLOBUS_ENV:
+        template_id = {
+            "sandbox": "33b6a241-bce4-4359-9c6d-09f88b3c9eef",
+            "integration": "e0c31fd1-663b-44e1-840f-f4304bb9ee7a",
+            "test": "0ebfd058-452f-40c3-babf-5a6b16a7b337",
+            "staging": "3029c3cb-c8d9-4f2b-979c-c53330aa7327",
+            "preview": "b2867dbb-0846-4579-8486-dc70763d700b",
+        }.get(GLOBUS_ENV, template_id)
+    return template_id
+
+
+def internal_native_client():
+    """
+    This is the client that represents the CLI itself (prior to templating)
+    """
+    template_id = _template_client_id()
+    return globus_sdk.NativeAppAuthClient(
+        template_id, app_name="Globus CLI (native client)"
+    )
+
+
+def _get_data_dir():
+    # get the dir to store Globus CLI data
+    #
+    # on Windows, the datadir is typically
+    #   ~\AppData\Local\globus\cli
+    #
+    # on Linux and macOS, we use
+    #   ~/.globus/cli/
+    #
+    # This is not necessarily a match with XDG_DATA_HOME or macOS use of
+    # '~/Library/Application Support'. The simplified directories for non-Windows
+    # platforms will allow easier access to the dir if necessary in support of users
+    if sys.platform == "win32":
+        # try to get the app data dir, preferring the local appdata
+        datadir = os.getenv("LOCALAPPDATA", os.getenv("APPDATA"))
+        if not datadir:
+            home = os.path.expanduser("~")
+            datadir = os.path.join(home, "AppData", "Local")
+        return os.path.join(datadir, "globus", "cli")
+    else:
+        return os.path.expanduser("~/.globus/cli/")
+
+
+def _ensure_data_dir():
+    dirname = _get_data_dir()
+    try:
+        os.makedirs(dirname)
+    except FileExistsError:
+        pass
+    return dirname
+
+
+def _get_storage_filename():
+    datadir = _ensure_data_dir()
+    return os.path.join(datadir, "storage.db")
+
+
+def _resolve_namespace():
+    env = GLOBUS_ENV if GLOBUS_ENV else "DEFAULT"
```

Comment:
todo: switch `"DEFAULT"` to `"production"` to make set/unset prod env behave the same

Reply:
I've just made this change and additionally added a layer to the namespacing here (explained in the comment).